src/webportal/src/app/job/job-view/fabric/job-detail/components/summary.jsx

```diff
@@ -96,82 +118,70 @@ export default class Summary extends React.Component {
       return;
     }
 
+    const HintItem = ({header, value}) => (
+      <div>
+        <span className={c(t.w4, t.dib)} style={{fontWeight: FontWeights.semibold}}>
+          {header}
+        </span>
+        <span>{value}</span>
+      </div>
+    );
+
     const state = getHumanizedJobStateString(jobInfo);
     if (state === 'Failed') {
-      const diag = jobInfo.jobStatus.appExitDiagnostics;
-      const code = jobInfo.jobStatus.appExitCode;
-      if (code === 177) {
-        // user code error
-        let userExitCode;
-        if (diag) {
-          let match = diag.match(/<Raw>\[ExitCode\]: (\d+)/);
-          if (match) {
-            userExitCode = parseInt(match[1], 10);
-          }
+      const result = [];
+      const spec = jobInfo.jobStatus.appExitSpec;
+      // error type
+      const type = spec && spec.type;
+      if (type) {
+        result.push(<HintItem key='type' header='Error Type:' value={spec.type} />);
+      }
+      // trigger task
+      const role = jobInfo.jobStatus.appExitTriggerTaskRoleName;
+      const idx = jobInfo.jobStatus.appExitTriggerTaskRoleIndex;
```

Comment:
Show appExitTriggerMessage?

Example:
      "applicationExitTriggerMessage" : "All Tasks completed and no ApplicationCompletionPolicy has ever been triggered: TotalTaskCount: 2, FailedTaskCount: 0",